test failures with Net-DNS-1.46

Lots of tests fails with Net-DNS-1.46

```
Test Summary Report
-------------------
./001basic/001self.t                  (Wstat: 256 (exited 1) Tests: 4 Failed: 1)
  Failed test:  3
  Non-zero exit status: 1
./003complex/010chars.t               (Wstat: 512 (exited 2) Tests: 4 Failed: 2)
  Failed tests:  2-3
  Non-zero exit status: 2
./004misc/019bad.t                    (Wstat: 256 (exited 1) Tests: 34 Failed: 1)
  Failed test:  32
  Non-zero exit status: 1
./003complex/017chaos.t               (Wstat: 768 (exited 3) Tests: 5 Failed: 3)
  Failed tests:  2-4
  Non-zero exit status: 3
./003complex/008compress.t            (Wstat: 512 (exited 2) Tests: 4 Failed: 2)
  Failed tests:  2-3
  Non-zero exit status: 2
./003complex/015rfc3597.t             (Wstat: 768 (exited 3) Tests: 5 Failed: 3)
  Failed tests:  2-4
  Non-zero exit status: 3
./007plugins/025metafo.t              (Wstat: 1024 (exited 4) Tests: 6 Failed: 4)
  Failed tests:  2-5
  Non-zero exit status: 4
./009monio/031_null_mon.t             (Wstat: 1024 (exited 4) Tests: 6 Failed: 4)
  Failed tests:  2-5
  Non-zero exit status: 4
./003complex/014naptr.t               (Wstat: 1280 (exited 5) Tests: 7 Failed: 5)
  Failed tests:  2-6
  Non-zero exit status: 5
./025nocookies/052nocookies.t         (Wstat: 256 (exited 1) Tests: 3 Failed: 1)
  Failed test:  2
  Non-zero exit status: 1
./020any/046any.t                     (Wstat: 512 (exited 2) Tests: 4 Failed: 2)
  Failed tests:  2-3
  Non-zero exit status: 2
./021caa/047caa.t                     (Wstat: 256 (exited 1) Tests: 3 Failed: 1)
  Failed test:  2
  Non-zero exit status: 1
./006root/021root.t                   (Wstat: 1024 (exited 4) Tests: 6 Failed: 4)
  Failed tests:  2-5
  Non-zero exit status: 4
./005tld/020tld.t                     (Wstat: 1024 (exited 4) Tests: 6 Failed: 4)
  Failed tests:  2-5
  Non-zero exit status: 4
./004misc/018misc.t                   (Wstat: 2560 (exited 10) Tests: 12 Failed: 10)
  Failed tests:  2-11
  Non-zero exit status: 10
./003complex/009broken.t              (Wstat: 2560 (exited 10) Tests: 15 Failed: 10)
  Failed tests:  2, 4, 7-14
  Non-zero exit status: 10
./010relax/036relax.t                 (Wstat: 2048 (exited 8) Tests: 10 Failed: 8)
  Failed tests:  2-9
  Non-zero exit status: 8
./003complex/007cname.t               (Wstat: 2560 (exited 10) Tests: 12 Failed: 10)
  Failed tests:  2-11
  Non-zero exit status: 10
./007plugins/023weighted.t            (Wstat: 2816 (exited 11) Tests: 13 Failed: 11)
  Failed tests:  2-12
  Non-zero exit status: 11
./002simple/003simple_errs.t          (Wstat: 3328 (exited 13) Tests: 15 Failed: 13)
  Failed tests:  2-14
  Non-zero exit status: 13
./007plugins/024dync.t                (Wstat: 3584 (exited 14) Tests: 16 Failed: 14)
  Failed tests:  2-15
  Non-zero exit status: 14
./027tcp_control/054tcp_control_ro.t  (Wstat: 2304 (exited 9) Tests: 14 Failed: 9)
  Failed tests:  2, 4-6, 8-11, 13
  Non-zero exit status: 9
./002simple/002noerr.t                (Wstat: 4352 (exited 17) Tests: 19 Failed: 17)
  Failed tests:  2-18
  Non-zero exit status: 17
./003complex/012aaaa.t                (Wstat: 4608 (exited 18) Tests: 20 Failed: 18)
  Failed tests:  2-19
  Non-zero exit status: 18
./012cname/038cname.t                 (Wstat: 3840 (exited 15) Tests: 17 Failed: 15)
  Failed tests:  2-16
  Non-zero exit status: 15
./003complex/005big.t                 (Wstat: 3072 (exited 12) Tests: 14 Failed: 12)
  Failed tests:  2-13
  Non-zero exit status: 12
./024cookies_max/051cookies_max.t     (Wstat: 3584 (exited 14) Tests: 16 Failed: 14)
  Failed tests:  2-15
  Non-zero exit status: 14
./003complex/006limits.t              (Wstat: 5376 (exited 21) Tests: 23 Failed: 21)
  Failed tests:  2-22
  Non-zero exit status: 21
./017extfile/043extfile.t             (Wstat: 768 (exited 3) Tests: 7 Failed: 3)
  Failed tests:  3-4, 6
  Non-zero exit status: 3
./022gdnsdctl/048gdnsdctl.t           (Wstat: 1792 (exited 7) Tests: 15 Failed: 7)
  Failed tests:  2, 4, 6, 8, 10, 12, 14
  Non-zero exit status: 7
./003complex/004delegations.t         (Wstat: 6912 (exited 27) Tests: 29 Failed: 27)
  Failed tests:  2-28
  Non-zero exit status: 27
./027tcp_control/056tcp_control_ctl.t (Wstat: 2304 (exited 9) Tests: 16 Failed: 9)
  Failed tests:  2, 4, 6, 8, 10-12, 14-15
  Non-zero exit status: 9
./003complex/013wild.t                (Wstat: 7936 (exited 31) Tests: 33 Failed: 31)
  Failed tests:  2-32
  Non-zero exit status: 31
./023cookies/050cookies.t             (Wstat: 5120 (exited 20) Tests: 22 Failed: 20)
  Failed tests:  2-21
  Non-zero exit status: 20
./007plugins/022trivial.t             (Wstat: 7936 (exited 31) Tests: 33 Failed: 31)
  Failed tests:  2-32
  Non-zero exit status: 31
./013clientsub/039clientsub.t         (Wstat: 7424 (exited 29) Tests: 31 Failed: 29)
  Failed tests:  2-30
  Non-zero exit status: 29
./008origin/026org_inc.t              (Wstat: 7680 (exited 30) Tests: 32 Failed: 30)
  Failed tests:  2-31
  Non-zero exit status: 30
./027tcp_control/055tcp_control_chal.t (Wstat: 2304 (exited 9) Tests: 16 Failed: 9)
  Failed tests:  2, 4-6, 8-9, 11, 13, 15
  Non-zero exit status: 9
./003complex/011random.t              (Wstat: 25600 (exited 100) Tests: 104 Failed: 100)
  Failed tests:  3-102
  Non-zero exit status: 100
./019many_zones/045many_zones.t       (Wstat: 25600 (exited 100) Tests: 102 Failed: 100)
  Failed tests:  2-101
  Non-zero exit status: 100
./014geoip/040basic.t                 (Wstat: 29184 (exited 114) Tests: 122 Failed: 114)
  Failed tests:  2-58, 63-119
  Non-zero exit status: 114
./009monio/028_sec_nolisten.t         (Wstat: 1024 (exited 4) Tests: 6 Failed: 4)
  Failed tests:  2-5
  Non-zero exit status: 4
./009monio/032_pri_tcp_nolisten.t     (Wstat: 1024 (exited 4) Tests: 6 Failed: 4)
  Failed tests:  2-5
  Non-zero exit status: 4
./009monio/027_pri_nolisten.t         (Wstat: 1024 (exited 4) Tests: 6 Failed: 4)
  Failed tests:  2-5
  Non-zero exit status: 4
./009monio/033_sec_tcp_nolisten.t     (Wstat: 1024 (exited 4) Tests: 6 Failed: 4)
  Failed tests:  2-5
  Non-zero exit status: 4
./016extmon/042extmon.t               (Wstat: 2048 (exited 8) Tests: 10 Failed: 8)
  Failed tests:  2-9
  Non-zero exit status: 8
./009monio/029_pri_listen.t           (Wstat: 1024 (exited 4) Tests: 7 Failed: 4)
  Failed tests:  2-5
  Non-zero exit status: 4
./009monio/035_sec_tcp_listen.t       (Wstat: 1024 (exited 4) Tests: 7 Failed: 4)
  Failed tests:  2-5
  Non-zero exit status: 4
./015zruntime/041simple.t             (Wstat: 2048 (exited 8) Tests: 10 Failed: 8)
  Failed tests:  2-9
  Non-zero exit status: 8
./009monio/030_sec_listen.t           (Wstat: 1024 (exited 4) Tests: 7 Failed: 4)
  Failed tests:  2-5
  Non-zero exit status: 4
./022gdnsdctl/049acmedns01.t          (Wstat: 20224 (exited 79) Tests: 89 Failed: 79)
  Failed tests:  3-15, 18-30, 32-44, 47-59, 61-73, 75-88
  Non-zero exit status: 79
./009monio/034_pri_tcp_listen.t       (Wstat: 1024 (exited 4) Tests: 7 Failed: 4)
  Failed tests:  2-5
  Non-zero exit status: 4
./011upthresh/037up_thresh.t          (Wstat: 2304 (exited 9) Tests: 14 Failed: 9)
  Failed tests:  2-8, 10, 12
  Non-zero exit status: 9
./018admin/044admin.t                 (Wstat: 9984 (exited 39) Tests: 60 Failed: 39)
  Failed tests:  3-6, 8-10, 12-14, 16-18, 20-22, 24-26, 28-30
                32-34, 36-38, 40-42, 44, 46, 48, 50, 52
                54, 57, 59
  Non-zero exit status: 39
Files=56, Tests=1155,  6 wallclock secs ( 1.06 usr  0.03 sys +  8.18 cusr  1.86 csys = 11.13 CPU)
Result: FAIL
make[3]: *** [Makefile:660: check-local] Error 1
```

I run the `001basic` test:
```
make check V=1 TRUN=001basic/*.t
...
make[2]: Entering directory '/home/ncopa/aports/community/gdnsd/src/gdnsd-3.8.2/t'
make  check-local
make[3]: Entering directory '/home/ncopa/aports/community/gdnsd/src/gdnsd-3.8.2/t'
Test data/outputs will be stored at /home/ncopa/aports/community/gdnsd/src/gdnsd-3.8.2/t/testout
if test "001basic/*.tx" != "x"; then \
	TOP_BUILDDIR=/home/ncopa/aports/community/gdnsd/src/gdnsd-3.8.2 TESTOUT_DIR=/home/ncopa/aports/community/gdnsd/src/gdnsd-3.8.2/t/testout TESTPORT_START=12345 /usr/bin/prove -v -f --norc -I. ./001basic/*.t; \
else \
	TOP_BUILDDIR=/home/ncopa/aports/community/gdnsd/src/gdnsd-3.8.2 TESTOUT_DIR=/home/ncopa/aports/community/gdnsd/src/gdnsd-3.8.2/t/testout TESTPORT_START=12345 /usr/bin/prove -v -f --norc -I. ./[0-9]*/*.t; \
fi
./001basic/001self.t .. 
1..4
ok 1 - use _GDT;
ok 2
Use of uninitialized value in numeric ne (!=) at /home/ncopa/aports/community/gdnsd/src/gdnsd-3.8.2/t/_GDT.pm line 734.
not ok 3

#   Failed test at ./001basic/001self.t line 16.
# IPv4 query failed: Packet mismatch! ID mismatch
# ---Received:
# ;; Response received from [127.0.0.1] 49 octets
# ;; HEADER SECTION
# ;;	id = 23789
# ;;	qr = 1	aa = 1	tc = 0	rd = 0	opcode = QUERY
# ;;	ra = 0	z  = 0	ad = 0	cd = 0	rcode  = NOERROR
# ;;	do = 0	co = 0
# ;;	qdcount = 1	ancount = 1
# ;;	nscount = 0	arcount = 0
# 
# ;; QUESTION SECTION (1 record)
# ;; ns1.example.com.	IN	A
# 
# ;; ANSWER SECTION (1 record)
# ns1.example.com.	86400	IN	A	192.0.2.42
# 
# ;; AUTHORITY SECTION (0 records)
# 
# ;; ADDITIONAL SECTION (0 records)
# 
# 
# ---Wanted:
# ;; HEADER SECTION
# ;;	id = undef
# ;;	qr = 1	aa = 1	tc = 0	rd = 0	opcode = QUERY
# ;;	ra = 0	z  = 0	ad = 0	cd = 0	rcode  = NOERROR
# ;;	do = 0	co = 0
# ;;	qdcount = 1	ancount = 1
# ;;	nscount = 0	arcount = 0
# 
# ;; QUESTION SECTION (1 record)
# ;; ns1.example.com.	IN	A
# 
# ;; ANSWER SECTION (1 record)
# ns1.example.com.	86400	IN	A	192.0.2.42
# 
# ;; AUTHORITY SECTION (0 records)
# 
# ;; ADDITIONAL SECTION (0 records)
# 
# 
# (Addr Limits: V4: 0 V6: 0)
ok 4
# Looks like you failed 1 test of 4.
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/4 subtests 

Test Summary Report
-------------------
./001basic/001self.t (Wstat: 256 (exited 1) Tests: 4 Failed: 1)
  Failed test:  3
  Non-zero exit status: 1
Files=1, Tests=4,  0 wallclock secs ( 0.01 usr  0.00 sys +  0.05 cusr  0.01 csys =  0.07 CPU)
Result: FAIL
```


I wonder if this is related this change in 1.46:

> Random ID cache moved from header->id to packet->encode.

https://metacpan.org/dist/Net-DNS/changes

--- a/t/_GDT.pm
+++ b/t/_GDT.pm
@@ -1056,6 +1056,7 @@ sub query_server {
         );
         send($sock, $qpacket_raw, 0);
         if($expected) {
+            $query->encode;  # generate a random ID, required with Net::DNS >= 1.46
             $expected->header->id($query->header->id);
             my $res_raw;
             recv($sock, $res_raw, 4096, 0);
@@ -1075,6 +1076,7 @@ sub query_server {
         }
 
         if($expected) {
+            $query->encode;  # generate a random ID, required with Net::DNS >= 1.46
             $expected->header->id($query->header->id);
             $size = _GDT->compare_packets($res->send($query), $expected, $limit_v4, $limit_v6, $wrr_v4, $wrr_v6);
         }

--- a/t/_GDT.pm
+++ b/t/_GDT.pm
@@ -1075,6 +1075,9 @@ sub query_server {
             $res->$k($ro->{$k})
         }
 
+        # Net::DNS <= 1.45 (r1982) did this internally
+        $query->edns->size($ro->{udppacketsize}) if defined $ro->{udppacketsize};
+
         if($expected) {
             $query->encode;  # generate a random ID, required with Net::DNS >= 1.46
             $expected->header->id($query->header->id);

